include/vinecopulib/bicop/implementation/class.ipp

```diff
@@ -657,19 +804,46 @@ Bicop::select(const Eigen::Matrix<double, Eigen::Dynamic, 2>& data,
   }
 }
 
-//! @brief Data manipulations for rotated families
-//!
-//! @param u \f$m \times 2\f$ matrix of data.
-//! @return The manipulated data.
-inline Eigen::Matrix<double, Eigen::Dynamic, 2>
-Bicop::cut_and_rotate(const Eigen::Matrix<double, Eigen::Dynamic, 2>& u) const
+//! adds an additional column if there's only one discrete variable;
+//! removes superfluous columns for continuous variables. 
+//! (continuous models only require two columns, discrete models always four)
+inline Eigen::MatrixXd
+Bicop::format_data(const Eigen::MatrixXd& u) const
 {
-  Eigen::Matrix<double, Eigen::Dynamic, 2> u_new(u.rows(), 2);
+  int n_disc = get_n_discrete();
+  if (n_disc == 0) {
+    return u.leftCols(2);
+  } else if ((get_n_discrete() != 1) | (u.cols() == 4)) {
+    return u;
+  }
+  Eigen::MatrixXd u_new(u.rows(), 4);
+  u_new.leftCols(2) = u.leftCols(2);
+  int disc_col = (var_types_[1] == "d");
```

Comment:
size_t ?

Reply:
Don't care actually, it's just because we used `size_t` everywhere. Just one thing: maybe you need a static cast or CRAN will complain...